homework/index.js

```diff
@@ -30,17 +36,91 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+  function renderError(error) {
+    const root = document.getElementById('root');
+    root.innerHTML = '';
+    createAndAppend('h1', root, { text: error.message, class: `alert-error` });
+  }
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepos(repo, div) {
+    div.innerHTML = '';
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: `_blank`,
     });
+    addRow(tbody, 'Description', repo.description);
+    addRow(tbody, 'Forks', repo.forks);
+    addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString('en-GB'));
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  function renderContributions(repo, ul) {
+    fetchJSON(repo.contributors_url)
+      .then(contributors => {
+        ul.innerHTML = '';
+        createAndAppend('h3', ul, { text: `Contributors:` });
```

Comment:
I think I already gave the feedback in week 1 that it is invalid to make an `h3` a child of an `li`.

Reply:
You're right I must have missed this one. Thank you